### ISSUE TITLE
How to Flatten dynamic field with parent document in Spring data Mongo DB

### ISSUE THREAD

**Issue by anshumt** [spring-projects/spring-boot#16067]
In my Spring boot project have a Document like so:

```
@Document(collection="AuditTable")
public class AuditTable {

    @Id
    private String id;

    private Map<String, String> properties;
```

where properties is a dynamic field i.e. it can take in as many different key-value pairs.

I use MongoRepository to store this value:

```
@Repository
public interface AuditTableRepo extends MongoRepository<AuditTable, String> {
}
```

Now when I store it in the Collection it looks like this:

<img width="705" alt="spring_mongo_ques" src="https://user-images.githubusercontent.com/20654747/53661451-52fd3980-3c2e-11e9-9155-1f8d33adff7a.png">

whereas I want it to look like this:

> "_id": "XYZ" 
> "_class": "XYZ" 
> "workCaseId":"12" 
> "taskName":"AUDIT" 
> "owner":"ANSHU" 
> "createdDate":"XYZ"

basically I want to flatten the properties to make the values it stores a part of the Root Document itself.

Any idea about how I should do this?

Thanks.

**Comment by mbhave** [comment]
Thanks for getting in touch, but it feels like this is a question that would be better suited to [Stack Overflow](http://stackoverflow.com/). As mentioned in [the guidelines for contributing](https://github.com/spring-projects/spring-boot/blob/master/CONTRIBUTING.adoc#using-github-issues), we prefer to use GitHub issues only for bugs and enhancements.
